packages/ethereum-contracts/contracts/apps/SuperTokenV1Library.sol

```diff
@@ -22,6 +22,16 @@ import {
  */
 library SuperTokenV1Library {
 
+    /**
+     * @notice Warms up the cache: get the host, cfa and ida from the token
+     * @dev Fixes the issue where expect revert fails because it expects this call to revert
+     * @param token The token used in the flow
+     */
+    function warmUpCache(ISuperToken token) external {
```

Comment:
This code smells,  please explain more.

Reply:
If we look at this test:
```
    function testRevertIfDecreaseFlowRateAllowanceUnderflows() public {
        vm.startPrank(alice);
        vm.expectRevert(IConstantFlowAgreementV1.CFA_ACL_NO_NEGATIVE_ALLOWANCE.selector);
        superToken.decreaseFlowRateAllowance(bob, 10);
        vm.stopPrank();
    }
```
it will fail if we do not warm up the cache with an error stating that no revert happened. this is because you are expecting the next call to revert and the next call is to: `getHost` inside of `_getHostAndCFA`.
if we remove the revert, it also fails, with the custom error.

the warming up of the cache is to prevent this. 
an alternative is to just execute a view function before and write a comment about this.

Reply:
I would prefer we don't have a special function in production code just for testing purpose. Instead, calling another function that incidentally warms up the catch sounds a clean workaround to me, so long as a comment in the test code explaining this "intended incidental warm up"

Reply:
e.g. create a flow just for testing